pkg/build/generator/generator.go

```diff
@@ -585,10 +583,8 @@ func (g *BuildGenerator) resolveImageStreamReference(ctx kapi.Context, from kapi
 	case "ImageStreamTag":
 		imageStreamTag, err := g.Client.GetImageStreamTag(kapi.WithNamespace(ctx, namespace), from.Name)
 		if err != nil {
-			glog.V(2).Infof("Error resolving ImageStreamTag reference %s in namespace %s: %v", from.Name, namespace, err)
-			if errors.IsNotFound(err) {
-				return "", err
-			}
+			err = fmt.Errorf("Error resolving ImageStreamTag reference %s in namespace %s: %v", from.Name, namespace, err)
```

Comment:
Not sure if NotFound is appropriate here, seems like something the server would return if the BuildConfig itself didn't exist. Perhaps BadRequest or UnprocessableEntity?